Docs: fix config file info in upgrade guide (#83273)
grafana/grafana: docs/sources/shared/upgrade/upgrade-common-tasks.md

## Upgrade Grafana

> The following sections provide instructions for how to upgrade Grafana based on your installation method. For more information on where to find configuration files, refer to [Configuration file location](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#configuration-file-location).

Reviewer: ```suggestion
The following sections provide instructions for how to upgrade Grafana based on your installation method. For more information on where to find configuration files, refer to [Configuration file location](/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#configuration-file-location).
```

### Debian

To upgrade Grafana installed from a Debian package (`.deb`), complete the following steps:

> 1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install_dir>/grafana.ini`.

Reviewer 2: ```suggestion
1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install_dir>/custom.ini`.
```

> 1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install_dir>/grafana.ini`.

Author: ```suggestion
1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install_dir>/custom.ini`.
```

Reviewer: 👀  I understood this like Isabel says.

Reviewer 2: After doing some research, @imatwawana you pointed it correctly that there is NO `custom.ini` in Linux. So in short its ONLY `grafana.ini` in Linux distributions

Kindly, make changes where needed in docs

Thank you again !! :slightly_smiling_face:

Author: The other instructions all follow this pattern, so they're fine. I'll go ahead and merge this! Thanks for all your help!

> 1. In your current installation of Grafana, save your custom configuration changes to a file named `/user/local/etc/grafana/grafana.ini`.

Reviewer: ```suggestion
1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install>/grafana.ini`.
```

> 1. In your current installation of Grafana, save your custom configuration changes to a file named `/user/local/etc/grafana/grafana.ini`.

Reviewer: ```suggestion
1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install_dir>/grafana.ini`.
```

To upgrade Grafana installed from the Grafana Labs APT repository, complete the following steps:

> 1. In your current installation of Grafana, save your custom configuration changes to a file named `/user/local/etc/grafana/grafana.ini`.

Reviewer: ```suggestion
1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install_dir>/grafana.ini`.
```

To upgrade Grafana installed using RPM or YUM complete the following steps:

> 1. In your current installation of Grafana, save your custom configuration changes to a file named `/user/local/etc/grafana/grafana.ini`.

Reviewer: ```suggestion
1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install_dir>/grafana.ini`.
```

To upgrade Grafana running in a Docker container, complete the following steps:

> 1. In your current installation of Grafana, save your custom configuration changes to the custom configuration file, `custom.ini` or `grafana.ini`.

Author: Can either of you @ppcano or @usmangt use the suggestion feature to suggest a specific update here. I've updated this a couple times now and I don't know what is wanted here.

> 1. Use Grafana [environment variables](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#override-configuration-with-environment-variables) to save your custom configurations; this is the recommended method. Alternatively, you can copy your configuration files manually by accessing the deployed container. 

Author: ```suggestion
1. Use Grafana [environment variables](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#override-configuration-with-environment-variables) to save your custom configurations; this is the recommended method. Alternatively, you can copy your configuration files manually by accessing the deployed container. 
```

> 1. Use Grafana [environment variables](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#override-configuration-with-environment-variables) to save your custom configurations; this is the recommended method. Alternatively, you can copy your configuration files manually by accessing the deployed container. 

Author: ```suggestion
1. Use Grafana [environment variables](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#override-configuration-with-environment-variables) to save your custom configurations; this is the recommended method. Alternatively, you can view your configuration files manually by accessing the deployed container. 
```

To upgrade Grafana installed on Windows, complete the following steps:

> 1. In your current installation of Grafana, save your custom configuration changes to a file named `/user/local/etc/conf/custom.ini`.

Reviewer: ```suggestion
1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install_dir>/conf/custom.ini`.
```

To upgrade Grafana installed on Mac, complete the following steps:

> 1. In your current installation of Grafana, save your custom configuration changes to a file named `/user/local/etc/conf/custom.ini`.

Reviewer: ```suggestion
1. In your current installation of Grafana, save your custom configuration changes to the custom configuration file, `custom.ini` or `grafana.ini`.
```

> 1. In your current installation of Grafana, save your custom configuration changes to the custom configuration file, `custom.ini` or `grafana.ini`.

Author: I'm confused at this comment. This whole section is about how to update the custom files for each OS. I don't understand what flow is being broken. I've updated based on @ppcano's feedback about where the custom files are found based on the Configure Grafana docs. If the information about where the custom files are found isn't correct, then that needs to be updated by someone who knows this for sure, but this section isn't about the default files.

The reason we have this content repeated in sections is so that someone can go straight to their OS and have the information there. People don't always necessarily read the information at the top of a page and we've set up this content this way to facilitate it for the user (vs just writing it once ourselves and then never again).

> 1. In your current installation of Grafana, save your custom configuration changes to the custom configuration file, `custom.ini` or `grafana.ini`.

Reviewer: > Reference example: see NOTE on this page beginning:
https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#configure-grafana

@usmangt , that doc page says:

> Depending on your OS, your custom configuration file is either the $WORKING_DIR/conf/defaults.ini file or 

This seems wrong, doesn't it?

I've created a [PR](https://github.com/grafana/grafana/pull/83169) to reference `custom.ini`. Please, check this out.
